pkg/apis/pipeline/v1beta1/pipelinerun_validation.go

```diff
@@ -21,72 +21,57 @@ import (
 	"fmt"
 
 	"github.com/tektoncd/pipeline/pkg/apis/validate"
-	"k8s.io/apimachinery/pkg/api/equality"
 	"knative.dev/pkg/apis"
 )
 
 var _ apis.Validatable = (*PipelineRun)(nil)
 
 // Validate pipelinerun
 func (pr *PipelineRun) Validate(ctx context.Context) *apis.FieldError {
-	if err := validate.ObjectMetadata(pr.GetObjectMeta()).ViaField("metadata"); err != nil {
-		return err
-	}
-	return pr.Spec.Validate(ctx)
+	errs := validate.ObjectMetadata(pr.GetObjectMeta()).ViaField("metadata")
+	return errs.Also(pr.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec"))
 }
 
 // Validate pipelinerun spec
-func (ps *PipelineRunSpec) Validate(ctx context.Context) *apis.FieldError {
-	if equality.Semantic.DeepEqual(ps, &PipelineRunSpec{}) {
-		return apis.ErrMissingField("spec")
-	}
```

Comment:
Is it handled by https://github.com/tektoncd/pipeline/pull/3279/files#diff-d03ea89d809deff286bf1459dc561b27R32 this change?

Reply:
Yeah, should have commented, it is handled by `// Check that one of PipelineRef and PipelineSpec is present` indeed